test/acvp_client.py

```diff
@@ -3,49 +3,64 @@
 # SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
 
 # ACVP client for ML-KEM
-#
-# Processes json files from
-# https://github.com/usnistgov/ACVP-Server/blob/master/gen-val/json-files
-#
+# See https://pages.nist.gov/ACVP/draft-celi-acvp-ml-kem.html and
+# https://github.com/usnistgov/ACVP-Server/tree/master/gen-val/json-files
 # Invokes `acvp_mlkem{lvl}` under the hood.
 
 import argparse
 import os
 import json
 import sys
 import subprocess
+import urllib.request
+from pathlib import Path
 
 # Check if we need to use a wrapper for execution (e.g. QEMU)
 exec_prefix = os.environ.get("EXEC_WRAPPER", "")
 exec_prefix = exec_prefix.split(" ") if exec_prefix != "" else []
 
-acvp_dir = "test/acvp_data"
-acvp_jsons = [
-    (
-        f"{acvp_dir}/acvp_v1.1.0.38_keyGen_prompt.json",
-        f"{acvp_dir}/acvp_v1.1.0.38_keyGen_expectedResults.json",
-    ),
-    (
-        f"{acvp_dir}/acvp_v1.1.0.39_keyGen_prompt.json",
-        f"{acvp_dir}/acvp_v1.1.0.39_keyGen_expectedResults.json",
-    ),
-    (
-        f"{acvp_dir}/acvp_v1.1.0.40_keyGen_prompt.json",
-        f"{acvp_dir}/acvp_v1.1.0.40_keyGen_expectedResults.json",
-    ),
-    (
-        f"{acvp_dir}/acvp_v1.1.0.38_encapDecap_prompt.json",
-        f"{acvp_dir}/acvp_v1.1.0.38_encapDecap_expectedResults.json",
-    ),
-    (
-        f"{acvp_dir}/acvp_v1.1.0.39_encapDecap_prompt.json",
-        f"{acvp_dir}/acvp_v1.1.0.39_encapDecap_expectedResults.json",
-    ),
-    (
-        f"{acvp_dir}/acvp_v1.1.0.40_encapDecap_prompt.json",
-        f"{acvp_dir}/acvp_v1.1.0.40_encapDecap_expectedResults.json",
-    ),
-]
+
+def download_acvp_files(version="v1.1.0.40"):
+    """Download ACVP test files for the specified version if not present."""
+    base_url = f"https://raw.githubusercontent.com/usnistgov/ACVP-Server/{version}/gen-val/json-files"
+
+    # Files we need to download for ML-KEM
+    files_to_download = [
+        "ML-KEM-keyGen-FIPS203/prompt.json",
+        "ML-KEM-keyGen-FIPS203/expectedResults.json",
+        "ML-KEM-encapDecap-FIPS203/prompt.json",
+        "ML-KEM-encapDecap-FIPS203/expectedResults.json",
+    ]
+
+    # Create directory structure
+    data_dir = Path(f"test/.acvp-data/{version}/files")
+    data_dir.mkdir(parents=True, exist_ok=True)
+
+    for file_path in files_to_download:
+        local_file = data_dir / file_path
+        local_file.parent.mkdir(parents=True, exist_ok=True)
+
+        if not local_file.exists():
+            url = f"{base_url}/{file_path}"
+            print(f"Downloading {file_path}...", file=sys.stderr)
+            try:
+                urllib.request.urlretrieve(url, local_file)
+                # Verify the file is valid JSON
+                with open(local_file, "r") as f:
+                    json.load(f)
+            except json.JSONDecodeError as e:
```

Comment:
Nit: `urllib.request.urlretrieve(url, local_file)` could be taken out of the scope of this `try ... except` block and get its own error message. It's not unthinkable that the URLs change or the server is down at some point.

Reply:
Ah, ignore. You have different `except` blocks anyway.

Reply:
It does have it's own except block? 
```
            except Exception as e:
                print(f"Error downloading {file_path}: {e}", file=sys.stderr)
                local_file.unlink(missing_ok=True)
                return False
```

What are you missing from that error message?

Reply:
It's ok, I considered two `try` blocks for better separation, but you're right that with the first block confinded ty JSON errors, it should work as intended.